packages/fast-check/src/arbitrary/letrec.ts

```diff
@@ -50,6 +57,154 @@ export type LetrecLooselyTypedTie = (key: string) => Arbitrary<unknown>;
  */
 export type LetrecLooselyTypedBuilder<T> = (tie: LetrecLooselyTypedTie) => LetrecValue<T>;
 
+/**
+ * Constraints to be applied on {@link letrec}
+ * @remarks Since 4.4.0
+ * @public
+ */
+export interface LetrecConstraints {
```

Comment:
Let's do that in a follow up PR so that we unlock this one:

We should import and export it from fast-check-default.ts so that end users can import it too.

It should be a type rather than an interface to fest with what we currently do for all others (to be confirmed, I have rechecked).